src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/BookingResourceIntTest.kt

```diff
@@ -212,14 +211,25 @@ class BookingResourceIntTest : ResourceTest() {
         .headers(setClientAuthorisation(listOf("ROLE_VIEW_PRISONER_DATA")))
         .exchange()
         .expectStatus().isOk
+        .expectBody().jsonPath("totalElements").isEqualTo(1)
     }
 
     @Test
-    @Disabled("this test fails - code update needed")
-    fun `returns 403 if prison is not in user caseload`() {
+    fun `returns empty list if prison is not in user caseload`() {
```

Comment:
Is this consistent with other endpoints? Wrong caseload and no override role feels like a "not authorised" scenario - 403. Or maybe even a 404 if that's what other endpoints do?

Reply:
yeah - had a long chat with Andy about it - because the work to determine if a user can view an offender (in this case via prisonId or bookingId or offenderNo) is done in the sql (@programmaticAuthorisation annotation), it was agreed this is acceptable.  Otherwise we would be going down a long road to fix all.  I am happy for the team to discuss this as I can see your point, but it's a difficult one... 
should we first test access and only then do the sql call?  If we do go down that route, we probably need to change all @programmaticAuthorisation annotations that Steve added on as part of checking authorisation as they all work in the same way. Looking in prison-api - there are 24 occurrences of this.